packages/dds/tree/src/core/tree/detachedFieldIndex.ts

```diff
@@ -35,15 +38,43 @@ import { IIdCompressor } from "@fluidframework/id-compressor";
  */
 export type ForestRootId = Brand<number, "tree.ForestRootId">;
 
+/**
+ * fake revision used to mark that the revision stored in a {@link DetachedFieldIndex} is not yet
+ * set after loading data from a summary
+ */
+export const fakeRevisionWhenNotSet = 33 as SessionSpaceCompressedId;
+
 /**
  * The tree index records detached field IDs and associates them with a change atom ID.
  */
 export class DetachedFieldIndex {
-	// TODO: don't store the field key in the index, it can be derived from the root ID
-	private detachedNodeToField: NestedMap<Major, Minor, ForestRootId> = new Map();
+	/**
+	 * A mapping from detached node ids to forest root ids and the revision that last created or modified the root.
+	 *
+	 * @remarks
+	 * undefined revisions are tolerated but any roots not associated with a revision must be disposed manually
```

Comment:
Which scenarios lead to undefined revisions being used?

Reply:
`applyTipChange` in the `SharedTreeMutableChangeEnricher` allows for undefined revisions but now that I'm looking at the usages, it seems like it's only because of rollbacks. Is there a reasonable revision we can use in that case?

Reply:
Okay, I think that's fine for now. I've made a note to clean that up in the future.